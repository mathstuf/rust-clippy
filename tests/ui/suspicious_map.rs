#![warn(clippy::suspicious_map)]
#![allow(clippy::redundant_closure)]

fn main() {
    let _ = (0..3).map(|x| x + 2).count();

    // This usage is OK because the `sum` side effect makes the `map` useful.
    let mut sum = 0;
    let _ = (0..3).map(|x| sum += x).count();

    // The linter is blind to path-based arguments however.
    let mut ext_sum = 0;
    let ext_closure = |x| ext_sum += x;
    let _ = (0..3).map(ext_closure).count();

    // The linter can see `FnMut` calls however.
    let mut ext_closure_inner = |x| ext_sum += x;
    let _ = (0..3).map(|x| ext_closure_inner(x)).count();
}
