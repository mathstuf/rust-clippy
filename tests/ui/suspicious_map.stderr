error: this call to `map()` won't have an effect on the call to `count()`
  --> $DIR/suspicious_map.rs:5:13
   |
LL |     let _ = (0..3).map(|x| x + 2).count();
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::suspicious-map` implied by `-D warnings`
   = help: make sure you did not confuse `map` with `filter` or `for_each`

error: this call to `map()` won't have an effect on the call to `count()`
  --> $DIR/suspicious_map.rs:14:13
   |
LL |     let _ = (0..3).map(ext_closure).count();
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: make sure you did not confuse `map` with `filter` or `for_each`

error: aborting due to 2 previous errors

